### DISCUSSION TITLE
Quick Question - is the latest python CSX interface build-able?

### DISCUSSION THREAD

**Discussion by BryceBuschman** [thliebig/openEMS-Project#247] [Q&A Usage] [answered]
Hi all, quick question - is the latest python CSX interface at https://github.com/thliebig/openEMS-Project.git currently build-able on  Windows 10 python 3.11? If not, is this something a naive user that knows very little about real C++ and python development (aka me) would be well equipped to fix? It seems the latest python interface on github is considerably ahead of the latest pre-built wheels. I'd like to access the additional functionality and am confined to a python development environment.

Build observations:
When building, I am getting an error (copied below), which may be the same as mentioned in the recent post #225 (original post). The build seems to 'almost work' and goes well until the CSXCAD/ParameterObjects.h file is not found. I found this file in openEMS-Project\CSXCAD\src. After copying the files from src into the CSXCAD folder, the build proceeded further. The next error indicated CSXCAD.lib could not be opened (also copied below). Unfortunately I did not see this file in the repository and had to stop my wild goose chase.

Thanks thliebig and many others who have no doubt spent significant time on this project. I am very appreciative of all your efforts. You have made the world a better place.

Error excerpt when building:
"C:\openEMS-Project\CSXCAD\python\CSXCAD\CSPrimitives.cpp(1257): fatal error C1083: Cannot open include file: 'CSXCAD/ParameterObjects.h': No such file or directory
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.41.34120\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2"

Error excerpt when building with contents of src copied to CSXCAD folder:
"LINK : fatal error LNK1181: cannot open input file 'CSXCAD.lib'"

**Comment by SengerM** [comment]
I was having issues installing in modern Linux distributions and I made a modification in the python module so it is pip-installable (the standard and recommended way for python packages). In case you want to give it a try, [here is my fork, branch `pip_installation`](https://github.com/SengerM/CSXCAD/tree/pip_installation/python). Not sure this will fix your problems, but it it works like a charm for me. Feel free to ask if yo need some guidance.

**Reply by BryceBuschman** [reply]
Thanks SengerM! Will take a look.

**Comment by thliebig** [comment, marked as answer]
I think to build openEMS and all dependencies on Windows is very very difficult. I always struggle with a new setup for weeks.
If you want to compile openEMS yourself I can only recommend to switch to Linux.

**Reply by BryceBuschman** [reply]
Thanks thliebig. This answers my question.
